CitadelExtras/HeadBob.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+WIP Headbobbing, weapon bobbing, and footsteps NOV 22.
+
+IMPORTANT: In order for headbobbing to work, the Y value "0.84f" for the player MUST be removed in the appropriate script.

```

Comment:
I'll have to look later but it might be cleaner to pull the screen shake portion out of MouseLookScript and put it in here as well so that there's no interdependency like this and all MainCamera localPosition changes happen in a single place.

Also would be more performant to have MouseLook's Update call this script's function after caching a reference in MouseLookScripts Start using GetComponent over there as each Update, even an empty one, has a hidden overhead of about 0.11ms from what I've seen in editor profiling on my machine anyways.